qemu/tests/chardev_add_port.py

```diff
@@ -0,0 +1,35 @@
+from virttest import error_context
+from virttest import utils_test
+from virttest import qemu_monitor
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Add existed virtio serial port and serial bus
+    1) Start guest with virtio-serial-port and virtio-serial-pci
+    2) execute":"qmp_capabilities"
+    3) Hot-plug existed port1
+    4) Hot plug existed virtio-serial-pci
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    if params['os_type'] == 'windows':
+        utils_test.qemu.windrv_check_running_verifier(
+            vm.wait_for_login(), vm, test, 'vioser', 300)
+
+    port = params.objects('serials')[1]
+    virtio_port = vm.devices.get(port)
+    pci_dev_id = virtio_port.params['bus'].split('.')[0]
+    pci_dev = vm.devices.get(pci_dev_id)
+    try:
+        pci_dev.hotplug(vm.monitor)
+    except qemu_monitor.QMPCmdError as e:
+        assert 'Duplicate' in e.data['desc'], 'Expected error is not duplicate pci device'
+    try:
+        virtio_port.hotplug(vm.monitor)
+    except qemu_monitor.QMPCmdError as e:
+        assert 'Duplicate' in e.data['desc'], 'Expected error is not duplicate virtio-serial device'
```

Comment:
`Expected error/Actual error` ?  The expect error is "Duplicate", if the actual error is not that, test fail.
@michiboo :
```
try:
    *hotplug*
except qemu_monitor.QMPCmdError as e:
    if 'Duplicate' not in e.data['desc']:
        test.fail(*****)
else:
    test.fail(*****)
```

Reply:
Normally, we plug virtio-serial-pci first, then the port. Since this is a negative test, i do not think the order matters. 
@nanliu-r Please help confirm and make the automation step match with the polarion test case, thanks!